app/javascript/components/mentoring/discussion/FileViewer.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState } from 'react'
+import { File } from '../../types'
+import { highlight } from '../../../utils/highlight'
+
+export const FileViewer = ({
+  language,
+  file,
+}: {
+  language: string
+  file: File
+}): JSX.Element => {
+  const [content, setContent] = useState('')
+
+  useEffect(() => {
```

Comment:
```suggestion
  useLayoutEffect(() => {
```

You probably want this to happen inline, and not "as a second render"